osu.Server.Queues.ScoreStatisticsProcessor/Processors/MedalAwarders/StatisticsMedalAwarder.cs

```diff
@@ -0,0 +1,80 @@
+﻿using JetBrains.Annotations;
+using System.Collections.Generic;
+using osu.Server.Queues.ScoreStatisticsProcessor.Models;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor.Processors.MedalAwarders
+{
+    [UsedImplicitly]
+    public class StatisticsMedalAwarder : IMedalAwarder
+    {
+        public bool RunOnFailedScores => false;
+
+        public IEnumerable<Medal> Check(IEnumerable<Medal> medals, MedalAwarderContext context)
+        {
+            foreach (var medal in medals)
+            {
+                if (checkMedal(medal, context))
+                    yield return medal;
+            }
+        }
+
+        private bool checkMedal(Medal medal, MedalAwarderContext context)
+        {
+            var score = context.Score;
+            var stats = context.UserStats;
+
+            return medal.achievement_id switch
```

Comment:
I'll let these switch statements pass for now, but reserve the right to convert them away from expression types in the future. Still not sold and don't think it helps that much with legibility.